flink-clients/src/test/java/org/apache/flink/client/program/PackagedProgramRetrieverImplTest.java

```diff
@@ -215,20 +215,6 @@ public void testCreateWithUserLibDir() throws FlinkException {
                 is(classpathProvider.getJobClassName()));
     }
 
-    @Test
-    public void testCreateWithoutUserLibDir() throws IOException, FlinkException {
```

Comment:
I could not fully understand why we need to remove the `testCreateWithoutUserLibDir` here. IIUC, it is used to verify the system classpath pickup.

Reply:
Not setting the user classpath makes the code fallback to the system classpath. We're not able to modify the system classpath after JVM is started already. Modifying `java.class.path` doesn't help here. The default system classpath contains multiple classes with main methods. That's why, the test would fail. The code path is, instead, covered by PackagedProgramRetrieverImplTest.testDeriveEntryClassInformationFromSystemClasspathExtractingTheJobClassFromThere`.